### PR TITLE
feat: LC-368 stats backend

### PR DESCRIPTION
**Pull request by typeWolffo**
## Jira issue(s)

https://selleolabs.atlassian.net/browse/LC-371
https://selleolabs.atlassian.net/browse/LC-375
https://selleolabs.atlassian.net/browse/LC-376
https://selleolabs.atlassian.net/browse/LC-377
https://selleolabs.atlassian.net/browse/LC-378

## Overview
- [x] API endpoint to provide statistics for student
- [x] Provide data for course statistics
- [x] Provide data for lesson statistics
- [x] Provide data for quiz statistics
- [x] Provide data for streak of using platform

### CODE REVIEW

File: apps/api/src/auth/auth.service.ts [hunk 1]
```diff
@@ -120,7 +121,7 @@ export class AuthService {
       const tokens = await this.getTokens(user);
       return tokens;
     } catch (error) {
-      throw new UnauthorizedException("Invalid refresh token");
+      throw new ForbiddenException("Invalid refresh token");
```
Comment (typeWolffo) [inline apps/api/src/auth/auth.service.ts:124]: this probably should fix our problem with infinite /refresh requests

File: apps/api/src/lessons/lessons.service.ts [hunk 2]
```diff
@@ -163,6 +166,10 @@ export class LessonsService {
       userId,
       true,
     );
+
+    let correctAnswers = 0;
+    let wrongAnswers = 0;
```
Comment (typeWolffo) [inline apps/api/src/lessons/lessons.service.ts:171]: @wielopolski any ideas on how to get this data in a better way?

Comment (wielopolski) [reply apps/api/src/lessons/lessons.service.ts:171]: maybe
```suggestion
    const correctAnswerCount = questionLessonItem.content.questionAnswers
  .filter(answer => Boolean(answer.isCorrect))
  .length;
  const wrongAnswerCount = totalQuestions - correctAnswerCount
```

**Review by piotr-pajak — APPROVED** [review]

**Review by wielopolski — APPROVED** [review]
LGTM 🍻 
I left suggestion

**Pull request merged**